corehq/apps/locations/views.py

```diff
@@ -392,10 +396,23 @@ def archive_location(request, domain, loc_id):
 @require_http_methods(['DELETE'])
 @can_edit_location
 def delete_location(request, domain, loc_id):
-    loc = Location.get(loc_id)
-    if loc.domain != domain:
-        raise Http404()
-    loc.full_delete()
+    try:
+        loc = Location.get(loc_id)
+        if loc.domain != domain:
+            raise Http404()
+        loc.full_delete()
+    except ResourceNotFound:
+        # Sometimes the couch and sql locations go out of sync and we can end up
+        # in a state where the couch doc is deleted and the sql doc still exists.
+        loc = SQLLocation.objects.prefetch_related(
+            'location_type').get(location_id=loc_id)
+        # delete the sql location anyway
+        loc.get_descendants(include_self=True).delete()
+        logger.error(
+            'Location with ID [{}] in domain [{}] was missing its couch doc '
+            'upon deletion. If this is recurring it might be worth checking '
+            'out any couch to postrgres issues.'.format(loc_id, domain)
+        )
```

Comment:
If SQL data is missing, you're kinda screwed, I think.  SQLLocations handle all the relationships (parentage, etc), and are pretty strict.  All bulk gets use SQL to get the ids and then `iter_docs` to get couch locations, so if you were missing SQLLocations, you'd still have a very hard time finding the couch locations.  This whole thing just  emphasizes how much we really need to get off couch locations.